test/uint-test.js

```diff
@@ -1,23 +1,113 @@
+}
+
+function makeTests(uIntType) {
+  describe(uIntType, function() {
+    let rippleType = ripple[uIntType];
+    let tests = fixtures[uIntType];
+
+    it('from_json().to_json()', function() {
+      tests['from_json().to_json()'].forEach(function(test) {
+        let result = rippleType.from_json(test.input);
+        assert.strictEqual(result.is_valid(), String(test.expected) !== 'null', `Validity check failed: ${test.input}`);
+        result = result.to_json();
+
+        if (test.expected === 'null') {
+          // XXX
+          // UInt160.to_json() returns NaN rather than null if input is invalid
+          assert.strictEqual(lodash.isNaN(result), true, resultError(test, result));
+        } else {
+          assert.strictEqual(result, test.expected, resultError(test, result));
+        }
+      });
+    });
+    it('from_json().to_bytes()', function() {
+      tests['from_json().to_bytes()'].forEach(function(test) {
+        let result = rippleType.from_json(test.input);
+        assert.strictEqual(result.is_valid(), String(test.expected) !== 'null', `Validity check failed: ${test.input}`);
+        assert.deepEqual(result.to_bytes(), test.expected, resultError(test, result));
+      });
+    });
+    it('from_number().to_json()', function() {
+      tests['from_number().to_json()'].forEach(function(test) {
+        let result = rippleType.from_number(test.input);
+        assert.strictEqual(result.is_valid(), String(test.expected) !== 'null', `Validity check failed: ${test.input}`);
+        result = result.to_json();
+
+        if (test.expected === 'null') {
+          // XXX
+          // UInt160.to_json() returns NaN rather than null if input is invalid
+          assert.strictEqual(lodash.isNaN(result), true, resultError(test, result));
+        } else {
+          assert.strictEqual(result, test.expected, resultError(test, result));
+        }
       });
-      it('should create 00000000000000000000000000000001 when called with 1', function () {
-        var val = UInt128.from_number(1);
-        assert.strictEqual(val.to_hex(), '00000000000000000000000000000001');
+    });
+    it('from_number().to_hex()', function() {
+      tests['from_number().to_hex()'].forEach(function(test) {
+        let result = rippleType.from_number(test.input);
+        assert.strictEqual(result.is_valid(), String(test.expected) !== 'null', `Validity check failed: ${test.input}`);
+        assert.strictEqual(result.to_hex(), test.expected, resultError(test, result));
       });
-      it('should create 000000000000000000000000FFFFFFFF when called with 0xFFFFFFFF', function () {
-        var val = UInt128.from_number(0xFFFFFFFF);
-        assert.strictEqual(val.to_hex(), '000000000000000000000000FFFFFFFF');
+    });
+    it('from_generic().to_*()', function() {
+      [
+      ['to_json', ...tests['from_json().to_json()']],
```

Comment:
Looks like the names in the fixtures should be "from_generic().to_*()"? I think it would be a bit simpler if this array was just `['to_json', 'to_bytes', 'to_hex']` and the test cases were looked up inside the loop.